### PR TITLE
feat(es/compat): Readonly and writeonly private field

### LINKED ISSUE #5599: Class properties pass should handle readonly on write-only fields
### Describe the bug

Currently the class properties pass swallows an error.

### Input code

```typescript
class Cl {
  #privateField = 0;

  set #privateFieldValue(newValue) {
    this.#privateField = newValue;
  }

  constructor() {
    expect(() => this.#privateFieldValue).toThrow(TypeError);
  }
}

const cl = new Cl();
```

### Config

```json
{
  "jsc": {
    "parser": {
      "syntax": "ecmascript"
    },
    "target": "es2021",
    "minify": {
      "compress": false,
      "mangle": false
    },
    "loose": false
  },
  "minify": false
}
```

### Playground link

https://play.swc.rs/?version=1.2.241&code=H4sIAAAAAAAAA22PwQrCQAxE7%2F2KgJf0UryXehH9guJ9WQNdWLpLklpF%2FHfjLj0o3pLMzBvioxOBY4RnA7DLHG5O6RwoXmGAfd%2FYVUi%2FlYuLC%2BFMaxnaEgXQKUj3S9hMvVleH5hPsygvXhPjlqR7Jq9o%2B3D4g6ktnaZx4rTi%2BMh0Yk7cVqhhCxR8rIX2DZr2Bt0BZ5baAAAA&config=H4sIAAAAAAAAA02LywqEMBAE%2F6XPHtRj%2FmYIE8mSF9M5rMj%2B%2BxpU8NbVRR340MMdaGJUG4t76fKFg%2Fos9BZbx29CF9u0j5vrvC6YkGOJYR%2BJr7mZknBBEvVUUrakN55xqpUvfMqL%2F63bUfaEAAAA

### Expected behavior

It should report an error.

babel output:
```js
"use strict";

function _classPrivateFieldInitSpec(obj, privateMap, value) { _checkPrivateRedeclaration(obj, privateMap); privateMap.set(obj, value); }

function _checkPrivateRedeclaration(obj, privateCollection) { if (privateCollection.has(obj)) { throw new TypeError("Cannot initialize the same private elements twice on an object"); } }

function _writeOnlyError(name) { throw new TypeError("\"" + name + "\" is write-only"); }

function _classPrivateFieldSet(receiver, privateMap, value) { var descriptor = _classExtractFieldDescriptor(receiver, privateMap, "set"); _classApplyDescriptorSet(receiver, descriptor, value); return value; }

function _classExtractFieldDescriptor(receiver, privateMap, action) { if (!privateMap.has(receiver)) { throw new TypeError("attempted to " + action + " private field on non-instance"); } return privateMap.get(receiver); }

function _classApplyDescriptorSet(receiver, descriptor, value) { if (descriptor.set) { descriptor.set.call(receiver, value); } else { if (!descriptor.writable) { throw new TypeError("attempted to set read only private field"); } descriptor.value = value; } }

var _privateField = /*#__PURE__*/new WeakMap();

var _privateFieldValue = /*#__PURE__*/new WeakMap();

class Cl {
  constructor() {
    _classPrivateFieldInitSpec(this, _privateFieldValue, {
      get: void 0,
      set: _set_privateFieldValue
    });

    _classPrivateFieldInitSpec(this, _privateField, {
      writable: true,
      value: 0
    });

    expect(() => (this, _writeOnlyError("#privateFieldValue"))).toThrow(TypeError);
  }

}

function _set_privateFieldValue(newValue) {
  _classPrivateFieldSet(this, _privateField, newValue);
}

const cl = new Cl();
```

### Actual behavior

_No response_

### Version

1.2.241

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Austaras**
**Description:**

This is the most useless feature I've ever wrote.

**BREAKING CHANGE:**

**Related issue (if exists):**
closes #5599

### CODE REVIEW

File: crates/swc_ecma_transforms_compat/src/es2022/class_properties/private_field.rs [hunk 1]
```diff
@@ -544,6 +574,8 @@ impl<'a> PrivateAccessVisitor<'a> {
             return (Expr::dummy(), None);
         }
 
+        if kind.is_writeonly() {}
```
Comment (jridgewell) [inline crates/swc_ecma_transforms_compat/src/es2022/class_properties/private_field.rs:577]: Leftover?

**Review by jridgewell — COMMENTED** [review]
Can you add a test case for the writing to a private method?

```js
class Foo {
  #method() {
  }

  test() {
    // methods are always read-only.
    this.#method = 1;
  }
}
```

**Comment by Austaras** [comment]
> Can you add a test case for the writing to a private method?
> 
> ```js
> class Foo {
>   #method() {
>   }
> 
>   test() {
>     // methods are always read-only.
>     this.#method = 1;
>   }
> }
> ```

I prefer not to, as there's already tsc test privateNameReadonly

**Review by jridgewell — DISMISSED** [review]
Ah, I didn't realize `privateNameReadonly.1.normal.js` was testing a private method. There doesn't seem to be a source file in our repo?

**Comment by kdy1** [comment]
@jridgewell It's in `crates/swc_ecma_parser/tests/tsc`

**Review by kdy1 — DISMISSED** [review]
swc-bump:
 - swc_ecma_transforms_compat

**Review by kdy1 — APPROVED** [review]

**Pull request merged**